kratos/modeler/cad_tessellation_modeler.h

```diff
@@ -0,0 +1,252 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Dagmawi Bekel
+//                   Ruben Zorrilla
+//
+
+#if !defined(KRATOS_CAD_TESSELLATION_MODELER_INCLUDED)
+#define KRATOS_CAD_TESSELLATION_MODELER_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "modeler.h"
+#include "utilities/nurbs_curve_tessellation.h"
+#include "geometries/brep_surface.h"
+#include "geometries/brep_curve_on_surface.h"
+#include "integration/triangle_gauss_legendre_integration_points.h"
+
+namespace Kratos {
+
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+ */
+class KRATOS_API(KRATOS_CORE) CadTessellationModeler : public Modeler {
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of CadTessellationModeler
+    KRATOS_CLASS_POINTER_DEFINITION(CadTessellationModeler);
+
+    typedef std::size_t IndexType;
+
+    typedef std::size_t SizeType;
+
+    typedef Node<3> NodeType;
+
+    typedef Point EmbeddedNodeType;
+
+    typedef PointerVector<NodeType> ContainerNodeType;
+
+    typedef PointerVector<EmbeddedNodeType> ContainerEmbeddedNodeType;
+
+    typedef BrepSurface<ContainerNodeType, ContainerEmbeddedNodeType> BrepSurfaceType;
+
+    typedef BrepCurveOnSurface<ContainerNodeType, ContainerEmbeddedNodeType> BrepCurveOnSurfaceType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    CadTessellationModeler() : Modeler()
+    {
+    }
+
+    /**
+     * @brief Constructor using a Model and Parameters
+     * @param rModel Reference of the Model
+     * @param ModelerParameters Parameters of the discretization
+     */
+    CadTessellationModeler(
+        Model& rModel,
+        Parameters ModelerParameters = Parameters())
```

Comment:
I'm not fully convinced about this, specially considering that this kind of utilities are understood to be (potentially) called from Python (note that in order to preserve the default parameter we will need to export it as a kwarg).

Hence I'm very much in favor of having a defaults getter method.